src/ibex_bluesky_core/callbacks/document_logger.py

```diff
@@ -0,0 +1,40 @@
+"""Logs all documents that the BlueSky run engine creates via a callback."""
+
+import json
+import os
+from pathlib import Path
+
+log_location = Path(os.path.join("C:\\", "instrument", "var", "logs", "bluesky", "raw_documents"))
+
+
+class DocLoggingCallback:
+    """Logs all documents under log_location, with the file name of their UID (.log)."""
+
+    def __init__(self) -> None:
+        """Initialise current_start_document and filename."""
+        self.current_start_document = None
+        self.filename = None
+
+    def __call__(self, name: str, document: dict) -> int:
```

Comment:
Does this actually need to return an `int`? I didn't think callbacks returned anything but maybe I'm wrong...